data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -411,29 +418,51 @@ private <T> Record<Event> getRecord(ConsumerRecord<String, T> consumerRecord, in
         return new Record<Event>(event);
     }
 
-    private void processRecord(final AcknowledgementSet acknowledgementSet, final Record<Event> record) {
+    private <T> void processRecord(final AcknowledgementSet acknowledgementSet, final Record<Event> record) {
         // Always add record to acknowledgementSet before adding to
         // buffer because another thread may take and process
         // buffer contents before the event record is added
         // to acknowledgement set
         if (acknowledgementSet != null) {
             acknowledgementSet.add(record.getData());
         }
+        long numRetries = 0;
+        final int retrySleepTimeMs = 100;
+        // Donot pause until half the poll interval time has expired
+        final long maxRetries = topicConfig.getMaxPollInterval().toMillis() / (2 * retrySleepTimeMs);
```

Comment:
These two variables (`maxRetries` and `retrySleepMs`) can be moved elsewhere in the class to clarify they are not so dynamic.

Perhaps make `maxRetries` a field, give it a clearer name (`maxRetriesOnException`) and set it in the constructor:

```
maxRetriesOnException = topicConfig.getMaxPollInterval().toMillis() / (2 * retrySleepTimeMs);
```

You can make a static field for the other

```
private static final int RETRY_ON_EXCEPTION_SLEEP_MS = 100
```